onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -139,10 +172,13 @@ def data(self, request, **kwargs):
             if should_paginate:
                 instances = self.paginate_queryset(instances)
 
+            cleaned_instance_fields = clean_data_fields(
+                TableauDataSerializer(instances, many=True).data, xform)
+
             csv_df_builder = CSVDataFrameBuilder(
                 xform.user.username, xform.id_string, include_images=False)
             data = csv_df_builder._format_for_dataframe(
-                DataInstanceSerializer(instances, many=True).data)
+                cleaned_instance_fields)
 
             return self._get_streaming_response(data)
```

Comment:
I am wondering whether we should implement `clean_data_fields()` as a generator function to limit the memory overhead it could have.  Seems we would put all data in memory for a form here, but it also seems that might have been the case with the previous implementation.

Reply:
Yes, that does make it more optimal.
I have changed that to a generator expression